### PR TITLE
#323 trailer icon for external reference links

### PR DESCRIPTION
**Pull request by ddelpiano**

### CODE REVIEW

File: js/components/interface/menu/MenuSingleItem.js [hunk 1]
```diff
@@ -102,7 +102,12 @@ class MenuSingleItem extends React.Component {
             <i className={item.icon}></i>
           </ListItemIcon>
           : undefined}
-        <span className="menu-item-label">{item.label}</span>
+        <span className="menu-item-label">
+          {item.label}
+          {item.trailerIcon
+            ? <i style={{ marginLeft: "5px" }} className={item.trailerIcon}></i>
```
Comment (filippomc) [inline js/components/interface/menu/MenuSingleItem.js:108]: Better remove the inline margin

**Review by filippomc — APPROVED** [review]
Do we really need this ad-hoc functionality? Why just not include the trailer icon in the label from the application? Or change the order of the icon that we already have?

**Comment by ddelpiano** [comment]
> Do we really need this ad-hoc functionality? Why just not include the trailer icon in the label from the application? Or change the order of the icon that we already have?

@filippomc to be honest I thought about that but the label is defined as a string in the json, are you thinking about editing the json before the menu it's rendered and then add the object there?

**Comment by filippomc** [comment]
> > Do we really need this ad-hoc functionality? Why just not include the trailer icon in the label from the application? Or change the order of the icon that we already have?
> 
> @filippomc to be honest I thought about that but the label is defined as a string in the json, are you thinking about editing the json before the menu it's rendered and then add the object there?

I'm already using react component labels to replace the label with an icon and it's working well:

```javascript
{
      label: <span><SvgIcon name='eye' /><sub style={{ marginLeft: 2 }}><i className="fa fa-angle-down" /></sub></span>,
      icon: "",
     ...
    }
```
If there are use cases with a pure json file then I agree we need the feature. In that case, I think it's to better remove inline css and add an example of use to the showcase.

**Comment by filippomc** [comment]
Other imporant  thing to consider is we are more and more needing non-font icons in the new application design so the usual fontawesome class approach can be a strong limitation

**Review by filippomc — CHANGES_REQUESTED** [review]
Suggested changes:

- remove inline css
- reconsider font-based approach to display the icon

**Comment by tarelli** [comment]
@ddelpiano waiting on you when you get a chance

**Comment by filippomc** [comment]
@ddelpiano I guess this is outdated after https://github.com/openworm/geppetto-client/pull/333

**Pull request closed without merging**